### PR TITLE
bpo-33944: Add site.py site-packages tracing in verbose mode

### PR DESCRIPTION
**Pull request by native-api**
https://bugs.python.org/issue33944

### CODE REVIEW

File: Lib/site.py [hunk 1]
```diff
@@ -87,6 +87,10 @@
 USER_BASE = None
 
 
+def _trace(message):
+    if sys.flags.verbose:
+        print(message,file=sys.stderr)
```
Comment (methane) [inline Lib/site.py]:
```suggestion
        print(message, file=sys.stderr)
```

File: Lib/site.py [hunk 2]
```diff
@@ -155,6 +159,7 @@ def addpackage(sitedir, name, known_paths):
     else:
         reset = False
     fullname = os.path.join(sitedir, name)
+    _trace("Processing .pth file: %r" % fullname)
```
Comment (methane) [inline Lib/site.py]:
```suggestion
    _trace(f"Processing .pth file: {fullname!r}")
```

**Comment by methane** [comment]
I didn't mean single place.  I meant changing all `%` usages.

**Comment by native-api** [comment]
> @methane: I didn't mean single place. I meant changing all `%` usages.

Well, the suggested change feature commits exactly what you suggested (I don't actually know if I can change it at all), and I didn't look any further, trusting your review.
Could you please do the task to completion? There may be yet other small things to fix or something.

**Comment by methane** [comment]
> and I didn't look any further, trusting your review.

Please don't trust me as such.  I hope contributors is not robot just applying suggested change.  I hope contributors understand and grow up by review.

> Could you please do the task to completion? There may be yet other small things to fix or something.

At a code level, looks good me.
But I'm not sure that adding more `PYTHON*` env vars is really good thing for this particular case...

Adding more and more envvar makes it harder to users read manual and find option they needed by their self.

File: Doc/using/cmdline.rst [hunk 3]
```diff
@@ -351,6 +351,11 @@ Miscellaneous options
    (filename or built-in module) from which it is loaded.  When given twice
    (:option:`!-vv`), print a message for each file that is checked for when
    searching for a module.  Also provides information on module cleanup at exit.
+
+   .. versionadded:: 3.8
```
Comment (remilapeyre) [inline Doc/using/cmdline.rst]:
```suggestion
   .. versionchanged:: 3.10
```

File: Lib/test/test_site.py [hunk 4]
```diff
@@ -310,6 +310,15 @@ def test_no_home_directory(self):
             mock_addsitedir.assert_not_called()
             self.assertFalse(known_paths)
 
+    def test_trace(self):
+        import io
+        message="bla-bla-bla"
```
Comment (remilapeyre) [inline Lib/test/test_site.py]:
```suggestion
        message = "bla-bla-bla"
```

File: Lib/test/test_site.py [hunk 5]
```diff
@@ -310,6 +310,15 @@ def test_no_home_directory(self):
             mock_addsitedir.assert_not_called()
             self.assertFalse(known_paths)
 
+    def test_trace(self):
+        import io
```
Comment (remilapeyre) [inline Lib/test/test_site.py]: Can you please move the import at the top of the file?

File: Lib/test/test_site.py [hunk 6]
```diff
@@ -310,6 +310,15 @@ def test_no_home_directory(self):
...
+        message="bla-bla-bla"
+        for verbose, out in (True, message+"\n"), (False, ""):
+            with mock.patch('sys.flags', mock.Mock(verbose=verbose)),\
```
Comment (remilapeyre) [inline Lib/test/test_site.py]:
```suggestion
            with mock.patch('sys.flags', mock.Mock(verbose=verbose)), \
```

File: Misc/NEWS.d/next/Library/2019-03-01-01-56-23.bpo-33944.-82Pkt.rst [hunk 7]
```diff
@@ -0,0 +1 @@
+Added site.py site-packages tracing in verbose mode
```
Comment (remilapeyre) [inline Misc/NEWS.d/next/Library/2019-03-01-01-56-23.bpo-33944.-82Pkt.rst]:
```suggestion
Added site.py site-packages tracing in verbose mode.
```

**Review by remilapeyre — COMMENTED** [review]
Thanks for the contribution @native-api !

File: Doc/using/cmdline.rst (hunk 3 above)
Comment (methane) [reply Doc/using/cmdline.rst]: s/versionadded/versionchanged/

File: Doc/using/cmdline.rst [hunk 8]
```diff
@@ -369,6 +369,11 @@ Miscellaneous options
    (filename or built-in module) from which it is loaded.  When given twice
    (:option:`!-vv`), print a message for each file that is checked for when
    searching for a module.  Also provides information on module cleanup at exit.
+
+   .. versionchanged:: 3.10
+      When running the :mod:`site` module, report the site-specific paths
```
Comment (remilapeyre) [inline Doc/using/cmdline.rst]:
```suggestion
      The :mod:`site` module reports the site-specific paths
```

**Review by remilapeyre — APPROVED** [review]

**Comment by native-api** [comment]
@methane Note that this also applies to all release branches since 3.7.

**Comment by native-api** [comment]
(Ignore my last (now deleted) message, confused this with my another PR)

**Pull request merged**